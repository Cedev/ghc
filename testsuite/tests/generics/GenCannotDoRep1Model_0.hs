{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE PolyKinds #-}

module CannotDoRep1Model_0 where

import GHC.Generics (Generic, Generic1)


-- We are not yet able to generate a generic representation for this difficult composition

data D f = D                                           deriving Generic1

data Compose2 f = Comp2 (f [D f])                      deriving Generic1

-- Nor for these larger compositions

-- data Compose3 f = Comp3 (f [[D f]])                    deriving Generic1
-- data Compose10 f = Comp10 (f [[[[[[[[[D f]]]]]]]]])    deriving Generic1
