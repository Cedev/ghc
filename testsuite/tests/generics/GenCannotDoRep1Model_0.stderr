
GenCannotDoRep1Model_0.hs:14:65: error:
    • Can't make a derived instance of ‘Generic1 Compose2’:
        Constructor ‘Comp2’ applies the parameter but the application
                            isn't of the the form (p (f p)) or (p c)
    • In the data declaration for ‘Compose2’
